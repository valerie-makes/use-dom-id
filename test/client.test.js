/**
 * @jest-environment jsdom
 */

import React from "react";
import MyApp from "./__MyApp";

import { resetRenderContext } from "../lib";
import { render, hydrate } from "react-dom";

import snapshots from "./__snapshots__/server.test.js.snap";

describe("client-side rendering", () => {
  let container;
  const logRender = jest.fn();

  beforeEach(() => {
    resetRenderContext();
    logRender.mockClear();

    container = document.createElement("div");
    document.body.innerHTML = "";
    document.body.appendChild(container);
  });

  it("should render correctly", () => {
    render(<MyApp logRender={logRender} />, container);
    expect(container.innerHTML).toMatchSnapshot();
    expect(logRender).toBeCalledTimes(1);
  });

  it("should hydrate correctly", () => {
    container.innerHTML = JSON.parse(
      snapshots["server-side rendering should render correctly 1"],
    );
    hydrate(<MyApp logRender={logRender} />, container);
    expect(container.innerHTML).toMatchSnapshot();
    expect(logRender).toBeCalledTimes(2);
  });
});
