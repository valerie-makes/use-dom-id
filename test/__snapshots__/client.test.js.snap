// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client-side rendering should render correctly 1`] = `"<form><label for=\\"id-client-0\\">Full Name</label><input id=\\"id-client-0\\" type=\\"text\\"><label for=\\"id-client-1\\">Email Address</label><input id=\\"id-client-1\\" type=\\"email\\"></form>"`;
